testsuite/features/support/env.rb

```diff
@@ -148,12 +148,19 @@ def capybara_register_driver
       print_server_logs
     end
   end
-  page.instance_variable_set(:@touched, false) if Capybara::Session.instance_created?
+  page.instance_variable_set(:@touched, false) if capybara_session_created?
```

Comment:
```suggestion
  page.instance_variable_set(:@touched, false) if !Capybara::Session.nil? && Capybara::Session.instance_created?
```

Reply:
* It's not really legible
* It makes us repeat ourselves
* It will not work, if I read this correctly:

Ruby Short-Circuit Evaluation
When Ruby evaluates expressions containing boolean operators, it uses short-circuit evaluation. With ||, if the expression on the left evaluates to true, it will return true. Otherwise, it will check if the expression on the right evaluates to true. If so, the expression returns true; otherwise, it will return false.

With &&, both the expression on the left and the expression on the right have to evaluate to true in order to return true. If either expression is false, it will return false

Reply:
:) Happy to hear that you prefer encapsulation ;P

Reply:
I knew you would say that :smile_cat: .

Like all good things, it depends...